add setup apim.sh

wlals761 opened the pull request:
- 최신 버젼의 릴리즈를 이용하여 함수앱 배포
- 배포한 펑션앱과 API Management 앱을 연결

wlals761 committed:
add ListMessages proxy function

wlals761 committed:
Merge branch 'main' into main

wlals761 committed:
Update NhnToast.Sms.csproj

wlals761 committed:
Update ListMessagesTests.cs

wlals761 committed:
Merge branch 'main' of https://github.com/devrel-kr/nhn-toast-notification-service-custom-connector

wlals761 committed:
Merge branch 'devrel-kr:main' into main

wlals761 committed:
Merge branch 'devrel-kr:main' into main

wlals761 committed:
Merge branch 'devrel-kr:main' into main

wlals761 committed:
Merge branch 'devrel-kr:main' into main

wlals761 committed:
Update README.md

wlals761 committed:
Merge branch 'devrel-kr:main' into main

wlals761 committed:
Merge branch 'devrel-kr:main' into main

wlals761 committed:
Merge branch 'devrel-kr:main' into main

wlals761 committed:
Merge branch 'devrel-kr:main' into main

wlals761 committed:
Merge branch 'devrel-kr:main' into main

wlals761 committed:
Create setup-apim.sh

wlals761 committed:
Revert "Update README.md"

This reverts commit d51542174091a7afaa064d2fb27d375a7bb07ea7.

wlals761 committed:
Update setup-apim.sh

wlals761 committed:
Merge branch 'devrel-kr:main' into feature/setup-apim

wlals761 committed:
Update setup-apim.sh

디렉토리명을 이용하여 펑션 앱 배포

wlals761 commented:
디렉토리 이름을 이용하여 펑션앱을 배포하도록 변경하였습니다

justinyoo commented on infra/setup-apim.sh:
```diff
@@ -0,0 +1,32 @@
+do
+    filename="$name-$version.zip"
+    smszip=$(echo $urls | jq --arg v $filename 'select(.name == $v) | .url' -r)
+    fncappName="fncapp-${directoryList[$directoryIndex]}-$environment-krc"
```
여기서 directoryList, directoryIndex 이 부분은 결국 `nt-sms` 또는 `nt-sms-verify` 가 될텐데, 이 값을 파라미터로 받아올 수는 없나요?

wlals761 replied:
수정하였습니다!

justinyoo reviewed (changes requested):
코멘트 남겨 뒀습니다!

wlals761 committed:
Update setup-apim.sh

파라미터로 함수앱 이름 받아오게 수정

wlals761 committed:
Update setup-apim.sh

justinyoo commented on infra/setup-apim.sh:
```diff
@@ -0,0 +1,30 @@
+
+set -e
+
+names=("SmsApp" "SmsVerificationApp")
```
이건 더이상 필요 없죠?

justinyoo commented on infra/setup-apim.sh:
```diff
@@ -0,0 +1,30 @@
+names=("SmsApp" "SmsVerificationApp")
+
+#'dev', 'test', 'prod', 'kdy', 'kms', 'lsw', 'pjm'
+environment=$1
+
+fncappSuffixes=($2 $3)
```
sufixes 값이 두개가 될지 세개가 될지 아니면 열 개가 될지 아무도 모릅니다. 따라서, 여기는 이렇게 받아오면 안되겠죠. 다른 형태로 받아와야 할 거예요.

예를 들자면 `sms,sms-verify,mms,sos` 같은 식으로 받아와서 이걸 콤마 기준으로 나눠서 여기서 배열로 처리한다든가 하는 식으로 해야 함.

justinyoo replied:
그리고 이 두 값은 외부에서 환경변수로 주입을 시킨다고 가정하고 진행해 보시겠어요? 그러니까, 이미 이 두 값은 이 스크립트를 실행시키기 전에 이미 `AZ_ENVIRONMENT_CODE`, `AZ_APP_SUFFIXES` 같은 식으로 설정이 되어 있는 상태라고 가정하는 거지.

justinyoo commented on infra/setup-apim.sh:
```diff
@@ -0,0 +1,30 @@
+do
+    filename="$name-$version.zip"
+    smszip=$(echo $urls | jq --arg v $filename 'select(.name == $v) | .url' -r)
+    fncappName="fncapp-${fncappSuffixes[$suffixIndex]}-$environment-krc"
```
마찬가지로 여기서도 맨 뒤의 `krc` 값 역시 환경변수를 통해 `AZ_LOCATION_CODE` 같은 식으로 이미 설정이 되어 있다고 가정하고 합시다.

justinyoo reviewed (changes requested):
코멘트 남겨 뒀어요

wlals761 committed:
Update setup-apim.sh

suffixes 값을 ','로 구분하여 배열처리
AZ_ENVIRONMENT_CODE, AZ_APP_SUFFIXES, AZ_LOCATION_CODE 값 외부에서 환경변수로 주입한다고 가정

wlals761 commented:
수정하였습니다!
-suffixes 값을 ','로 구분하여 배열처리
-AZ_ENVIRONMENT_CODE, AZ_APP_SUFFIXES, AZ_LOCATION_CODE 값 외부에서 환경변수로 주입한다고 가정

wlals761 committed:
Update setup-apim.sh

파라미터 값 수정
펑션 앱 세팅 업데이트 추가

wlals761 committed:
Update setup-apim.sh

변수 수정

wlals761 committed:
Update setup-apim.sh

변수이름, 파라미터 값 수정

wlals761 committed:
Update setup-apim.sh

변수 수정

wlals761 committed:
Update setup-apim.sh

wlals761 commented:
변수이름, 파라미터 값 수정하였습니다

justinyoo reviewed (approved):
수고하셨습니다!
